File: civitasRUBY/lib/test_p4.rb
#encoding: utf-8

require_relative "civitas_juego"
require_relative "vista_textual"
require_relative "dado"

module Civitas
  class TestP4
    
    def self.main
      
      juego=CivitasJuego.new( ["Pepe","Juan","Marta"] )
      vista = VistaTextual.new
      controlador= Controlador.new(juego,vista)
      #Dado.instance.debug(true)
      
      #juego.jugadorAEspeculador(0) #Pepe será especulador
      #juego.jugadorAEspeculador(2) #Marta será especuladora
      
      controlador.juega
    end
  end
  
  Civitas::TestP4.main
end

File: civitasJAVA/src/civitas/testP4.java
package civitas;

import java.util.Arrays;
import java.util.ArrayList;

import JuegoTexto.Controlador;
import JuegoTexto.VistaTextual;


public class testP4 {
    
     public static void main(String[] args) {
        CivitasJuego juego = new CivitasJuego(new ArrayList<>(Arrays.asList("Pepe","Juan","Marta")));
        VistaTextual vista = new VistaTextual();
        Controlador controlador = new Controlador(juego,vista);
        
        //juego.jugadorAEspeculador(0); //Pepe  pasa a ser especulador
        //juego.jugadorAEspeculador(2); //Marta pasa a ser especuladora
        
        //Dado.getInstance().setDebug(true);
        controlador.juega();
     }
}